pcsx2/Recording/InputRecording.cpp

```diff
@@ -220,28 +239,32 @@ wxString InputRecording::RecordingModeTitleSegment()
 void InputRecording::SetToRecordMode()
 {
 	state = InputRecordingMode::Recording;
-	virtualPads[CONTROLLER_PORT_ONE]->SetReadOnlyMode(false);
-	virtualPads[CONTROLLER_PORT_TWO]->SetReadOnlyMode(false);
+	pads[CONTROLLER_PORT_ONE].virtualPad->SetReadOnlyMode(false);
+	pads[CONTROLLER_PORT_TWO].virtualPad->SetReadOnlyMode(false);
 	inputRec::log("Record mode ON");
 }
 
 void InputRecording::SetToReplayMode()
 {
 	state = InputRecordingMode::Replaying;
-	virtualPads[CONTROLLER_PORT_ONE]->SetReadOnlyMode(true);
-	virtualPads[CONTROLLER_PORT_TWO]->SetReadOnlyMode(true);
+	pads[CONTROLLER_PORT_ONE].virtualPad->SetReadOnlyMode(true);
+	pads[CONTROLLER_PORT_TWO].virtualPad->SetReadOnlyMode(true);
 	inputRec::log("Replay mode ON");
 }
 
 void InputRecording::SetFrameCounter(u32 newGFrameCount)
 {
-	if (newGFrameCount > startingFrame + (u32)g_InputRecording.GetInputRecordingData().GetTotalFrames())
+	const u32 endFrame = startingFrame + inputRecordingData.GetTotalFrames();
+	if (newGFrameCount >= endFrame)
```

Comment:
hmm, why the change to `>=` does `GetTotalFrames` return something different now? (havn't gotten to that point in the review yet) or does this resolve a bug of some sort?